### PR TITLE
[binary] Use Xcode built Info.plist's

### PR DESCRIPTION
**Pull request by paulb777**
Use Xcode built Info.plist's instead of self-generated ones when building binary distributions.

This will produce a more accurate Info.plist and be less likely to cause problems with App Store or other validation.

`CarthageUtils.generatePlistContents` is now used only for the placeholder Firebase framework.

The test results match current `main` (Outstanding bug with Resource directory structure)

More context for Googlers at b/318008447

#no-changelog

### CODE REVIEW

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift [hunk 1]
```diff
@@ -639,33 +635,7 @@ struct FrameworkBuilder {
           "\(framework): \(error)")
       }
 
-      // Move any privacy manifest-containing resource bundles into the
```
Comment (paulb777) [inline ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:642]: No functional change here. Just refactored to a function.

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift [hunk 2]
```diff
@@ -380,10 +380,6 @@ struct FrameworkBuilder {
       }
       umbrellaHeader = umbrellaHeaderURL.lastPathComponent
     }
-    // Add an Info.plist. Required by Carthage and SPM binary xcframeworks.
```
Comment (paulb777) [inline ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:383]: Remove Info.plist generation.

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift [hunk 3]
```diff
@@ -688,25 +658,23 @@ struct FrameworkBuilder {
...
+      let infoPlist = frameworkPath.appendingPathComponent("Info.plist").resolvingSymlinksInPath()
+      let infoPlistDestination = platformFrameworkDir.appendingPathComponent("Info.plist")
       let fatBinaryDestination = platformFrameworkDir.appendingPathComponent(framework)
       do {
         try fileManager.copyItem(at: fatBinary, to: fatBinaryDestination)
       } catch {
         fatalError("Could not copy fat binary to framework directory for \(framework): \(error)")
       }
+      do {
+        try fileManager.copyItem(at: infoPlist, to: infoPlistDestination)
```
Comment (paulb777) [inline ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:674]: copy the Info.plist from the xcodebuild invocation.

File: ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift (hunk 1 above)
Comment (andrewheard) [reply ReleaseTooling/Sources/ZipBuilder/FrameworkBuilder.swift:642]: LGTM. This helps with readability.

**Review by andrewheard — APPROVED** [review]
Thanks, Paul! This approach looks a lot less error-prone to me.

**Pull request merged**